npm_and_yarn/lib/dependabot/npm_and_yarn/package_name.rb

```diff
@@ -12,25 +12,29 @@ class PackageName
       # - https://github.com/npm/npm-user-validate
       # - https://github.com/npm/validate-npm-package-name
       PACKAGE_NAME_REGEX = %r{
-          \A                                          # beginning of string
-          (?=.{1,214}\z)                              # enforce length (1 - 214)
-          (@(?<scope>                                 # capture 'scope' if present
-            (?=[^\.])                                 # reject leading dot
-            [a-z0-9\-\_\.\!\~\*\'\(\)]+               # URL-safe characters
-          )\/)?
-          (?<name>                                    # capture package name
-            [a-z0-9\-\_\.\!\~\*\'\(\)]+               # URL-safe characters
+        \A                                          # beginning of string
+        (?=.{1,214}\z)                              # enforce length (1 - 214)
+        (@(?<scope>                                 # capture 'scope' if present
+          [a-z0-9\-\_\.\!\~\*\'\(\)]+               # URL-safe characters in scope
+        )\/)?                                       # scope must be followed by slash
+        (?<name>                                    # capture package name
+          (?(<scope>)                               # if scope is present
+            [a-z0-9\-\_\.\!\~\*\'\(\)]+             # scoped names can start with any URL-safe character
+          |                                         # if no scope
+            [a-z0-9\-\!\~\*\'\(\)]                  # non-scoped names cannot start with . or _
+            [a-z0-9\-\_\.\!\~\*\'\(\)]*             # subsequent characters can be any URL-safe character
           )
-          \z                                          # end of string
-      }xi                                             # multi-line/case-insensitive
+        )
+        \z                                          # end of string
+      }xi # multi-line/case-insensitive
 
       TYPES_PACKAGE_NAME_REGEX = %r{
           \A                                          # beginning of string
           @types\/                                    # starts with @types/
           ((?<scope>.+)__)?                           # capture scope
           (?<name>.+)                                 # capture name
           \z                                          # end of string
-      }xi                                             # multi-line/case-insensitive
```

Comment:
@raj-meka :This diff should not be there in the code. Request you to remove it.

Reply:
set it to original